core/tee/tee_svc_cryp.c

```diff
@@ -78,6 +79,11 @@ struct tee_cryp_obj_secret {
 #define ATTR_OPS_INDEX_BIGNUM     1
     /* Convert to/from value attribute depending on direction */
 #define ATTR_OPS_INDEX_VALUE      2
+    /* Convert to/from curve25519 attribute depending on direction */
+#define ATTR_OPS_INDEX_25519      3
```

Comment:
Why is this needed? All other algorithms use bignum and value.

Reply:
> Why is this needed? All other algorithms use bignum and value.

The X(Ed)25519 curve operations arguments are  32-byte octet strings as defined in
[RFC 7748, pg 7](https://datatracker.ietf.org/doc/html/rfc7748#page-7). Also from TEE Internal Core API Specification v1.2, Table Table 6-15, `TEE_ATTR_X25519_*` is a binary octet string (while other asym crypto attrs are bignum). The underlying implementation of libtomcrypt for Curve25519 operations does not use bignums (AFAIK most of the library implementations follow this convention). The X25519 byte level encoding is `little-endian` according to the `RFC-7748`, and if I am not mistaken for bignums it is big-endian. If we go the bignum route, we should then add conversions from bignum to appropriate binary format before calling the `X25519` ops.

Reply:
Makes sense, thanks for the details. Let's keep this new define.